yapapi/executor/_smartq.py

```diff
@@ -215,7 +216,16 @@ def current_item(self) -> Optional[Item]:
         """The most-recent queue item that has been fetched to be processed by this consumer."""
         return self._fetched.data if self._fetched else None
 
+    def finish(self):
+        self._finished = True
+
+    @property
+    def finished(self):
+        return self._finished
+
     async def __anext__(self) -> Handle[Item]:
+        if self._finished:
+            raise StopAsyncIteration()
         val = await self._queue.get(self)
         self._fetched = val
```

Comment:
Can `self._finished` be `True` after `await`?

Reply:
I think I don't understand this question... please eli5

Reply:
Resolved offline